### PR TITLE
upgrade: cleanup the incomplete state file

### PR DESCRIPTION
**Pull request by Vicente-Cheng**
**IMPORTANT: Please do not create a Pull Request without creating an issue first.**

**Problem:**
The incomplete `state.yaml` sometimes causes the upgrade to fail on the OS upgrading stage.

**Solution:**
Remove the incomplete `state.yaml`. We are good at default configuration.

**Related Issue:**
https://github.com/harvester/harvester/issues/4526

**Test plan:**
1. Install Harvester v1.1.1
2. Upgrade to v1.1.2 and check the `/run/cos/initramfs/cos-state/state.yaml` would be like following
```
# Autogenerated file by elemental client, do not edit

date: "2023-09-13T08:31:42Z"
state:
    active:
        source: dir:///tmp/tmp.01deNrXNEC
        label: COS_ACTIVE
        fs: ext2
    passive: null
```
4. upgrade to latest version (with this patch) and check the `/run/cos/initramfs/cos-state/state.yaml` would be like following
```
# Autogenerated file by elemental client, do not edit

date: "2023-09-23T17:59:24Z"
...
state:
    label: COS_STATE
    active:
        source: dir:///run/rootfsbase
        label: COS_ACTIVE
        fs: ext2
    passive:
        source: dir:///run/rootfsbase
        label: COS_PASSIVE
        fs: ext2
...
```

### CODE REVIEW

File: package/upgrade/upgrade_node.sh [hunk 1]
```diff
@@ -531,6 +555,9 @@ upgrade_os() {
   # The trap will be only effective from this point to the end of the execution
   trap clean_up_tmp_files EXIT
 
+  # we would like to clean up the incomplete state.yaml to avoid the issue of https://github.com/harvester/harvester/issues/4526
+  cleanup_incomplete_state_file
```
Comment (bk201) [inline package/upgrade/upgrade_node.sh]: suggest moving to a later stage, we can skip it if there is not need to upgrade OS.

Comment (Vicente-Cheng) [reply package/upgrade/upgrade_node.sh]: Updated! thanks for the suggestion

File: package/upgrade/upgrade_node.sh [hunk 2]
```diff
@@ -4,6 +4,28 @@ SCRIPT_DIR="$( cd "$( dirname "${BASH_SOURCE[0]}" )" &> /dev/null && pwd )"
 
 source $SCRIPT_DIR/lib.sh
 UPGRADE_TMP_DIR=$HOST_DIR/usr/local/upgrade_tmp
+STATE_DIR=/run/initramfs/cos-state
```
Comment (bk201) [inline package/upgrade/upgrade_node.sh]:
```suggestion
STATE_DIR=$HOST_DIR/run/initramfs/cos-state
```

Comment (Vicente-Cheng) [reply package/upgrade/upgrade_node.sh]: Good catch! Fixed!

**Review by ibrokethecloud — APPROVED** [review]
LGTM. since the issue is hard to reproduce I can confirm there is no impact to existing upgrade path.

**Review by bk201 — APPROVED** [review]
LGTM! Thanks.

**Review by starbops — APPROVED** [review]
Went through the test plan:

After upgrading to v1.1.2 from v1.1.1, the label is missing from the `state.yaml` file.

```
harvester-lcmsb:~ # cat !$
cat /run/initramfs/cos-state/state.yaml
# Autogenerated file by elemental client, do not edit

date: "2023-09-27T05:19:04Z"
state:
    active:
        source: dir:///tmp/tmp.iybWwbAUhf
        label: COS_ACTIVE
        fs: ext2
    passive: null
```

After upgraded to the custom build containing the fix, the `state.yaml` is in the correct format and the original one is backed up.

```
harvester-lcmsb:~ # cat /run/initramfs/cos-state/state.yaml
# Autogenerated file by elemental client, do not edit

date: "2023-09-27T09:05:51Z"
state:
    label: COS_STATE
    active:
        source: dir:///tmp/tmp.Cn64BgE8hR
        label: COS_ACTIVE
        fs: ext2
    passive:
        label: COS_PASSIVE
harvester-lcmsb:~ # cat /run/initramfs/cos-state/state.yaml.bak 
# Autogenerated file by elemental client, do not edit

date: "2023-09-27T05:19:04Z"
state:
    active:
        source: dir:///tmp/tmp.iybWwbAUhf
        label: COS_ACTIVE
        fs: ext2
    passive: null
```

LGTM, thanks!

**Comment by bk201** [comment]
@mergifyio backport v1.2

**Pull request merged**